src/ComboBox/ComboBox.svelte

```diff
@@ -269,29 +267,25 @@
           if (key === 'Enter') {
             open = !open;
 
-            if (highlightedIndex > -1 && highlightedIndex !== selectedIndex) {
-              selectedIndex = highlightedIndex;
+            if (
+              highlightedIndex > -1 &&
+              filteredItems[highlightedIndex]?.id !== selectedId
+            ) {
               open = false;
-              highlightedIndex = -1;
-              if (filteredItems[selectedIndex]) {
-                inputValue = filteredItems[selectedIndex].text;
-                selectedItem = filteredItems[selectedIndex];
-                selectedId = filteredItems[selectedIndex].id;
+              if (filteredItems[highlightedIndex]) {
```

Comment:
We can use `highlightedIndex` because `highlightedIndex = -1;` was moved to line 288.